src/codegen/unwinding.cpp

```diff
@@ -582,7 +582,9 @@ BoxedDict* getLocals(bool only_user_visible, bool includeClosure) {
             }
 
             for (const auto& p : cf->location_map->names) {
-                if (p.first[0] == '!')
+                if (p.first[0] == '!' && (only_user_visible
+                                          || !(p.first == PASSED_GENERATOR_NAME || p.first == PASSED_CLOSURE_NAME
+                                               || p.first == CREATED_CLOSURE_NAME)))
```

Comment:
I utterly failed to parse this conditional :(  I think this could be made easier to understand, but I think this is also evidence that these names should start with `#` and not `!` -- `!` are supposed to be irgen-internal names.  Would you mind trying switching the names and seeing if anything breaks?  If something does it's probably not worth hunting down unless you're curious.